### PR TITLE
#4072 remote patient lookup via qr code

### LINKED ISSUE #4072: Remote patient searching via QR Code
Child of #4066
Prerequisite: https://github.com/sussol/msupply/issues/8840

## Is your feature request related to a problem? Please describe.

As a user, I would like to be able to use a scanned QR code to search against online patient records (when a local search has returned no results), so that I don't need to manually enter search criteria into the app when dispensing a vaccine.

## Describe the solution you'd like
Given:
- The app has successfully retrieved a QR code, and 
- A local search has been performed with the QR code and no results were found
  - A search should be performed against the mSupply server patient API using the QR code
  - If there is a match, the user should be taken to Step 2 of the vaccine dispensing flow, with the relevant patient details prepopulated
  - If there is no match, the user should be told the patient couldn't be found, and be taken back to the vaccine dispensing page to either try again or enter details manually
 
## Implementation

TBD

## Describe alternatives you've considered

N/A

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #4072 

## Change summary

- Adds remote patient searching by QR code (needs https://github.com/sussol/msupply/pull/8914 to actually work)
- Adds crude validation to reject QR codes that are longer than `50` chars long (sample Vanuatu National ID we have seen is 21 characters long)
- Small side quest to remove the extra `getAuthorizationHeader` method in `SearchForm`

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Have a patient that is not available locally but is available remotely, scan a QR code corresponding to that patient -> should be taken to step 2 of the patient vaccination page with remote patient details pre-populated

### Related areas to think about
Might need to look into different parsers for different countries at some point (instead of storing whole thing, depending on what their National ID QR codes look like...). Either coded in mobile or some kind of store pref in Desktop?

### CODE REVIEW

File: src/widgets/Tabs/PatientSelect.js [hunk 1]
```diff
@@ -155,7 +168,19 @@ const PatientSelectComponent = ({
...
     } else {
-      ToastAndroid.show('No patient found via QR code', ToastAndroid.LONG);
+      // Do a remote search
+      withLoadingIndicator(async () => {
+        try {
+          const remotePatient = await lookupRemotePatient({ barcode: data });
+          if (remotePatient.length) {
+            selectPatient(remotePatient[0]);
+          } else {
+            throw new Error('No patient found via QR code');
```
Comment (joshxg) [inline src/widgets/Tabs/PatientSelect.js:178]: Think this needs to be localised since we're using it for user feedback?

**Review by joshxg — APPROVED** [review]
Badass

Just the localised strings!

File: src/widgets/Tabs/PatientSelect.js (hunk 1 above)
Comment (katherine-sussol) [reply src/widgets/Tabs/PatientSelect.js]: Thanks, yeah probs a good idea! 228d105

**Review by joshxg — APPROVED** [review]
🎉

**Pull request merged**